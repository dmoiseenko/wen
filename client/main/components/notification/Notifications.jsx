import React from 'react';
import PropTypes from 'prop-types';

import NotificationContainer from './Notification.container';


function Notifications({ notifications }) {
  return (
    <div className="notifications">
      {
        notifications.map(notification =>
          <NotificationContainer
            key={notification.id}
            notification={notification}
          />,
        )
      }
    </div>
  );
}

Notifications.propTypes = {
  notifications: PropTypes.array.isRequired,
};

export default Notifications;
