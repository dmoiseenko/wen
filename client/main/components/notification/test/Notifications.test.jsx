import React from 'react';
import { shallow } from 'enzyme';

import Notifications from '../Notifications';


it('should render', () => {
  const props = {
    notifications: [
      {
        id: '2',
      }
    ]
  };

  const wrapper = shallow(
    <Notifications {...props} />
  );

  expect(wrapper.node).toMatchSnapshot();
});
