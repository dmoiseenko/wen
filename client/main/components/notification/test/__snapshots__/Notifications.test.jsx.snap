// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="notifications"
>
  <Connect(Notification)
    notification={
      Object {
        "id": "2",
      }
    }
  />
</div>
`;
